### ISSUE TITLE
AzurermAdUsers.guest_accounts unusable for large AD

### ISSUE THREAD

**Issue by james-stocks** [inspec/inspec-azure#198]
Customer is running this test and reports that it always fails:

```
  describe azurerm_ad_users do
    its('guest_accounts') { should eq [] }
  end
```

## Possible Solution

The customer has a very large AD and believes the problem is due to the FilterTable loading _all_ user data then filtering in memory.
The `guest_accounts` method makes use of FilterTable filtering when it could possibly use a faster explicit query for guest accounts.

Current code:

```
def guest_accounts
    @guest_accounts ||= where(userType: 'Guest').mails
  end
```

**Comment by r-fennell** [comment]
Does the Resource & FilterTable sucessfully initialise? 
It's important to know if the problem is "there are too many users to load at once" or "there are too many entries in the FilterTable to performantly filter by".

If the problem is the first scenario, the suggested fix would be to 
1. Increase the available memory of the machine executing the control, and/or
2. Allow the Resource constructor to accept a search query to limit resultsets.

If the problem is the second scenario
1. Modify the `guest_accounts` method to query the API with a search/filter instead of the FilterTable, or
2. Allow the Resource constructor to accept a search query to limit resultsets.

Most suggestions above assuming some search or filter functionality is supported by Azure's Graph RBAC API.
If this is not the case, an alternative method may be to "stream" users via paganation and perform filtering client-side, dismissing any which do not match and minimizing memory impact.

In the case where an organisation wants to load all users (and the number of users they have is substantial), the only work around will be memory increases.

**Comment by trickyearlobe** [comment]
We have a separate repo we're sharing with UBS to track issues... I'll get some data in an issue there and link it (since we can't really give them access to raise issues on **this** repo.

Anecdotally, the resource runs for several mins before barfing.

**Comment by trickyearlobe** [comment]
Customer confirmed they have >150k accounts and the scan runs out of memory before it produces any results.

**Comment by trickyearlobe** [comment]
Customer actually has ~400k accounts in dev/test and > 500k accounts in Prod.

**Comment by rmoles** [comment]
Hi @trickyearlobe, I just wanted to check, did the PR listed above solved this issue?

**Comment by rmoles** [comment]
This issue has been resolved by #227
